flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -26,52 +26,69 @@
 import java.sql.SQLException;
 import java.sql.Statement;
 
-import org.apache.flink.api.common.io.NonParallelInput;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import org.apache.flink.api.common.io.DefaultInputSplitAssigner;
 import org.apache.flink.api.common.io.RichInputFormat;
 import org.apache.flink.api.common.io.statistics.BaseStatistics;
+import org.apache.flink.api.java.io.RangeInputSplit;
+import org.apache.flink.api.java.io.GenericRow;
 import org.apache.flink.api.java.tuple.Tuple;
 import org.apache.flink.configuration.Configuration;
 import org.apache.flink.core.io.GenericInputSplit;
 import org.apache.flink.core.io.InputSplit;
 import org.apache.flink.core.io.InputSplitAssigner;
-import org.apache.flink.types.NullValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * InputFormat to read data from a database and generate tuples.
  * The InputFormat has to be configured using the supplied InputFormatBuilder.
  * 
+ * Remark: split (if set) works only if the split column is numeric
+ * 
  * @param <OUT>
  * @see Tuple
  * @see DriverManager
  */
-public class JDBCInputFormat<OUT extends Tuple> extends RichInputFormat<OUT, InputSplit> implements NonParallelInput {
-	private static final long serialVersionUID = 1L;
+public class JDBCInputFormat extends RichInputFormat<GenericRow, InputSplit> {
 
+	private static final long serialVersionUID = 1L;
 	private static final Logger LOG = LoggerFactory.getLogger(JDBCInputFormat.class);
 
 	private String username;
 	private String password;
 	private String drivername;
 	private String dbURL;
-	private String query;
+	private String queryTemplate;
 	private int resultSetType;
 	private int resultSetConcurrency;
 
 	private transient Connection dbConn;
 	private transient Statement statement;
 	private transient ResultSet resultSet;
-
-	private int[] columnTypes = null;
+	
+	private String splitColumnName;
+	private long max;
+	private long min;
+	private long fetchSize;
+	
+	private boolean hasNext = true;
+	
+	private static final String BETWEEN = "(%s BETWEEN %s AND %s)";
+	public static final String CONDITIONS = "$CONDITIONS";
 
 	public JDBCInputFormat() {
 	}
 
 	@Override
 	public void configure(Configuration parameters) {
+		//called once per inputFormat (on open)
+		try {
+			establishConnection();
```

Comment:
I think I forgotto include the new connection management when I closed the previous branch..